File: 0125-valid-palindrome/0125-valid-palindrome.dart
class Solution {
  bool isPalindrome(String s) {
    final lowered = s.toLowerCase().replaceAll(RegExp(r'[^a-z0-9]'), '');

    if (lowered.length <= 1) {
      return true;
    }

    int left = 0;
    int right = lowered.length - 1;
    while (lowered[left] == lowered[right]) {
      if (left + 1 == right) {
        return true;
      }

      left++;
      right--;

      if (left == right) {
        return true;
      }
    }
    return false;
  }
}

File: 0125-valid-palindrome/0125-valid-palindrome.kt
class Solution {
    fun isPalindrome(s:String) : Boolean {
        val s = s.lowercase().filter { it.isLetterOrDigit() }
        var left = 0
        var right = s.length - 1

        while (left < right) {
            if (s[left] != s[right]) {
                return false
            }

            left += 1
            right -= 1
        }

        return true
    }
}
